### DISCUSSION TITLE
Loading images but timing is not consistent

### DISCUSSION THREAD

**Discussion by liadJB** [bonsai-rx/bonsai#1686] [Q&A]
I am trying to load images with a 0.16s interval between them (workflow attached)
[ShowImages.zip](https://github.com/bonsai-rx/bonsai/files/14385307/ShowImages.zip)
A few example images are also attached here.
[images.zip](https://github.com/bonsai-rx/bonsai/files/14385319/images.zip)
When running it over a long period of time (in my case, showing 3000 images), we can detect with the photodiode that same frames are dropping. They are logged but apparently not shown on the screen (we can see that from the photdiode trace)
Is there a way to prevent it from happening, or making it more stable, or am I doing something not optimally in the script?

Thanks
Liad

**Comment by bruno-f-cruz** [comment]
Heya, 
Without spending too much time with the workflow, two obvious solutions to try are:
1. Update packages to the latest version. If i am seeing this correctly alot of packages you are using are >2 years old. Some of which have suffered significant performance improvements since.
2. Load all assets at once at the start of the workflow instead of loading them 1-by-1 each time you present a stimuli.

Hope it helps
Bruno

**Reply by liadJB** [reply]
Thanks for the response.

1. I tried to update Bonsai, but when it updates older scripts all expression transforms are removed. Is there a way to update without losing parts of the scripts? 

2. Could you please show an example how to load the images at the start and show the, randomly as I do in the script?

Thanks
Liad

**Reply by bruno-f-cruz** [reply]
Heya!

1. I am able to use the latest version of all packages, except the NIDAQ one (probably because I don't have the API installed on my PC). Find attached the environment i am using to open your workflow, your original workflow (which mostly compiles except for the nidaq node, worst case scenario just replace it with the latest one in the toolbox) and a slightly more updated workflow that gets rid of a lot of deprecated nodes. You can find a tutorial on how to use [environments in the docs](https://bonsai-rx.org/docs/articles/environments.html)

2. I think you already asked this question a few months ago :P (https://github.com/orgs/bonsai-rx/discussions/1643). It is always a good idea to search for answers in old discussions first. It will save you time waiting for an answer and it will also save other people's time from duplicating answers over and over again. If you don't find what you are looking for or you have further questions, don't hesitate to ask for help in the original post :) !

[LoadImages.zip](https://github.com/bonsai-rx/bonsai/files/14393333/LoadImages.zip)

Hope it helps,
B

**Comment by liadJB** [comment]
Thanks for answering.

1) will try that. In case it fails is there a way to role back?

2) Sorry, but the solution there didn't work. I will answer it there so it
is in one place.

**Reply by bruno-f-cruz** [reply]
You can version control your code. Otherwise, no, as the workflow and bonsai will need to be modified.

**Reply by liadJB** [reply]
Hi,

I am trying to update to the new Bonsai. but for some reason whenever I load any of my old scripts, they are updated okay. But, then bonsai crashes without an error message.

During update, and also when I try to install the Numerics package manually, I get this error:

![image](https://github.com/bonsai-rx/bonsai/assets/48135253/d44be15a-7256-4d80-80dd-f38d33fdd05e)

I explicitly get an error says numerics is not installed, when I tried to open your texture loading solution from https://github.com/orgs/bonsai-rx/discussions/1643

Would you have any idea for resolving this error?

I really appreciate your help,

Liad

**Reply by bruno-f-cruz** [reply]
Instead of updating, can you try a clean install with the new bonsai version and only then update the packages you want? I am wondering if the version of your executable is just too old to allow a straight update.

**Reply by liadJB** [reply]
Managed to update successfully, but only by installing and downloading the new packages manually.
